pkg/controller/tas/resource_flavor.go

```diff
@@ -146,15 +186,13 @@ func (r *rfReconciler) Reconcile(ctx context.Context, req reconcile.Request) (re
 		r.tasCache.Delete(kueue.ResourceFlavorReference(req.NamespacedName.Name))
 	}
 	if flv.Spec.TopologyName != nil {
-		if r.tasCache.Get(kueue.ResourceFlavorReference(flv.Name)) == nil {
-			topology := kueuealpha.Topology{}
-			if err := r.client.Get(ctx, types.NamespacedName{Name: string(*flv.Spec.TopologyName)}, &topology); err != nil {
-				return reconcile.Result{}, err
-			}
-			levels := utiltas.Levels(&topology)
-			tasInfo := r.tasCache.NewTASFlavorCache(kueue.TopologyReference(topology.Name), levels, flv.Spec.NodeLabels)
-			r.tasCache.Set(kueue.ResourceFlavorReference(flv.Name), tasInfo)
+		topology := kueuealpha.Topology{}
+		if err := r.client.Get(ctx, types.NamespacedName{Name: string(*flv.Spec.TopologyName)}, &topology); err != nil {
+			return reconcile.Result{}, err
 		}
+		levels := utiltas.Levels(&topology)
+		tasInfo := r.tasCache.NewTASFlavorCache(kueue.TopologyReference(topology.Name), levels, flv.Spec.NodeLabels)
+		r.tasCache.Set(kueue.ResourceFlavorReference(flv.Name), tasInfo)
```

Comment:
Only set new if the nodeLabels, or levels are changed, to avoid throwing away the workload usage otherwise.

Reply:
Done